src/obj.rs

```diff
@@ -698,15 +683,18 @@ impl<'a> Parser<'a> {
 
     loop {
       match sliced(&self.peek()) {
```

Comment:
I don't think the `sliced` call is needed any more. Also, it might make sense to an a `fn eof(&self) -> bool` to `lex.rs` instead of using `peek` like this. Then this loop can be:

```
while !self.lexer.eof() {
  ...
}
```